src/citrine/_rest/resource.py

```diff
@@ -46,6 +48,49 @@ def access_control_dict(self) -> dict:
         }
 
 
+GEMDSelf = TypeVar('GEMDSelf', bound='GEMDResource')
+
+
+class GEMDResource(Resource[GEMDSelf]):
+    """A reference to a resource by UID."""
+
+    @classmethod
+    def build(cls, data: dict) -> GEMDSelf:
+        """Convert a raw, nested dictionary into Objects."""
+        if "context" in data and len(data) == 2:
+            def _inflate(x):
+                return DictSerializable.class_mapping[x["type"]].build(x)
+            key = next(k for k in data if k != "context")
+            idx = make_index([_inflate(x) for x in data["context"] + [data[key]]])
+            lst = [idx[k] for k in idx]
+            substitute_objects(lst, idx, inplace=True)
+
+            root = _inflate(data[key])
+            if root in idx:  # It was a link
+                return idx[root]
+            else:  # It was an object, but it won't be densely linked
+                return idx[root.to_link()]
```

Comment:
Support Material History-like serializations.  Handles both the return structure from the `material_history` route as well as the GEMD denormalized format.